### PR TITLE
run interceptRequest async

### PR DESCRIPTION
**Pull request by steffyP**
Issue: resolving the environment variable still sends the request to AWS, because reconfiguring the endpoint is running async, and is not finished when the `getParameter SSM` is called, resulting in error:
```
aws: [1] { UnrecognizedClientException: The security token included in the request is invalid.
    at Request.extractError (/path/ls-lambda/node_modules/aws-sdk/lib/protocol/json.js:52:27)
    at Request.callListeners (/path/ls-lambda/node_modules/aws-sdk/lib/sequential_executor.js:106:20)
    at Request.emit (/path/ls-lambda/node_modules/aws-sdk/lib/sequential_executor.js:78:10)
   ....

'Cannot resolve serverless.yml: Variables resolution errored with:\n  - Cannot resolve variable at "provider.environment.LAMBDA_STAGE": The security token included in the request is invalid.',
```
This is triggered by the following configuration for `LAMBDA_STAGE`:
```
provider:
  name: aws
  runtime: nodejs18.x
  stage: dev
  region: eu-west-1
  memorySize: 512
  timeout: 60
  iam:
    role:
      statements:
        - Effect: 'Allow'
          Resource: '*'
          Action:
            - 'lambda:InvokeFunction'
            ....
  environment:
    LAMBDA_STAGE: ${ssm:/${opt:stage, self:provider.stage}/lambda/common/LAMBDA_STAGE}
```

* It turns out, that in this case (for resolving the `environment` variable) the intercept-hook `interceptRequest` is triggered
   * the function `interceptRequest` is a patching [aws.request](https://github.com/serverless/serverless/blob/main/lib/plugins/aws/provider.js#L1683-L1697), which is already`async`. 
   * changing function to `async interceptRequest`,  and adding `await` for the aws-reconfiguration. 
   * added `init` hook that will also await the aws-reconfiguration
 
* changed the default for `BUCKET_MARKER_LOCAL` to hot-reload

### CODE REVIEW

File: spec/integration/integration.spec.js [hunk 1]
```diff
@@ -3,11 +3,45 @@
...
+const params = {
+  Name: '/dev/lambda/common/LAMBDA_STAGE',
+  Type: 'String',
+  Value: 'my-value',
+  Overwrite: true
+};
 
 describe('LocalstackPlugin', () => {
 
   beforeEach( () => {
```
Comment (whummer) [inline spec/integration/integration.spec.js:29]: nit: I think we could also await the `.promise()` on the SSM API call here: (same functionality, but would be slightly shorter)
```
  beforeEach( async () => {
    await ssm.putParameter(params).promise();
    this.service = services.createService({});
  });
```

File: src/index.js [hunk 2]
```diff
@@ -574,6 +566,10 @@ class LocalstackPlugin {
    */
   async reconfigureAWS() {
     if(this.isActive()) {
+      if(this.reconfigured_endpoints){
```
Comment (whummer) [inline src/index.js:569]: nit: we usually use `camelCase` in JavaScript/TypeScript code - could be changed to `this.reconfiguredEndpoints`

**Review by whummer — APPROVED** [review]
Great set of changes @steffyP ! 🚀 Kudos for digging into this deep rabbit hole. I think with this approach we now have a solid solution that enables async logic (to resolve the target endpoint (IPv4/IPv6 switch) properly in an async way) and proper initialization of the plugin with the `initialize` hook. 👍

File: spec/integration/integration.spec.js (hunk 1 above)
Comment (steffyP) [reply spec/integration/integration.spec.js:29]: thanks for the hint 🙏  i was actually struggling to set this up correctly.

**Pull request merged**